src/Tempest/HttpClient/src/Driver/Psr18Driver.php

```diff
@@ -28,17 +28,15 @@ public function __construct(
 
     public function send(Request $request): Response
     {
+        $psrRequest = $this->convertTempestRequestToPsrRequest($request);
+
         return $this->convertPsrResponseToTempestResponse(
-            $this->sendRequest($request)
+            $this->sendRequest($psrRequest)
         );
     }
 
-    public function sendRequest(Request|RequestInterface $request): ResponseInterface
+    public function sendRequest(RequestInterface $request): ResponseInterface
```

Comment:
`Request|RequestInterface` is necessary to be PSR compliant.

Reply:
I am very certain that is not true. `Psr\Http\Client\ClientInterface`  has an send method with this signature 
`public function sendRequest(RequestInterface $request): ResponseInterface;`. The `Request` is a `Tempest\Http\Request` which doesn't seem like it has anything to do with PSR.

The phpstan error actually said this, the method was required to be `sendRequest(RequestInterface $request): ResponseInterface` and this actually shouldn't accept `Tempest\Http\Request` which we did provide and this caused an error.

Reply:
It could actually be that we allowed `Request|RequestInterface` before, then refactored some stuff, and now only `RequestInterface` is needed. Part of the alpha phase 😅 If all tests are green and PHPStan is as well, then I'm fairly certain we're ok to drop `Request`

Reply:
Naw, we have to keep this in unless we are going to change some design decisions.

Originally I had it as a PSR only client, but @brendt wanted us to be able to send Tempest request objects as well. This change was made per his request [here](https://github.com/tempestphp/tempest-framework/pull/157#pullrequestreview-1912845989).